grifts.go

```diff
@@ -49,11 +49,63 @@ func middlewareGrift(a *App) {
 }
 
 func printMiddleware(a *App) {
-	fmt.Printf("-> %s\n", a.Name)
-	fmt.Printf("%v\n", a.Middleware.String())
-	for _, x := range a.children {
-		printMiddleware(x)
+	printMiddlewareByRoute(a)
+}
+
+func printMiddlewareByRoute(a *App) {
+	mws := map[string]string{}
+	for _, r := range a.Routes() {
+		if mws[r.App.Name] == "" {
+			pname := ""
+			if parent := getParentApp(r.App.root, r.App.Name); parent != nil {
+				pname = parent.Name
+			}
+
+			mws[r.App.Name] = r.App.Middleware.String()
+			if mws[pname] != mws[r.App.Name] {
+				fmt.Printf("-> %s\n", r.App.Name)
+				printMiddlewareStackWithIndent(mws[r.App.Name])
+			} else {
+				fmt.Printf("-> %s has same middleware stack as %v\n", r.App.Name, pname)
+			}
+		}
+		for k := range r.App.Middleware.skips {
+			mw := strings.Split(k, funcKeyDelimeter)[0]
+			h := strings.Split(k, funcKeyDelimeter)[1]
+			if h == r.HandlerName {
+				ahn := strings.Split(r.HandlerName, "/")
+				hn := ahn[len(ahn)-1]
+				fmt.Printf("-> %s %s (by %s)\n", r.Method, r.Path, hn)
+				s := strings.Replace("\n"+mws[r.App.Name], "\n"+mw, "", 1)
+				printMiddlewareStackWithIndent(s)
+			}
+		}
+	}
+}
+
+func getParentApp(r *App, name string) *App {
+	if r == nil {
+		return nil
+	}
+	for _, x := range r.children {
+		if x.Name == name {
+			return r
+		}
+		if len(x.children) > 0 {
+			if ret := getParentApp(x, name); ret != nil {
+				return ret
+			}
+		}
+	}
+	return nil
+}
+
+func printMiddlewareStackWithIndent(s string) {
+	if s == "" {
+		s = "[none]"
 	}
+	s = strings.Replace(s, "\n", "\n   ", -1)
```

Comment:
```suggestion
	s = strings.Replace(s, "\n", "\n\t", -1)
```

Reply:
OK, I agree with using tab even though it takes five more spaces and can generate line wrap on 80 col terminal like me, anyway that is fancier!